### PR TITLE
add the form for the comment and likes

### PR DESCRIPTION
**Pull request by X-Elie-X**
### in this PR I implemented the below:
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database.
- Create forms to perform the following functions:
Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @author = User.find_by(id: params[:user_id])
+    @post = @author.posts.new(post_params)
+    @post.likes_counter = 0
+    @post.comments_counter = 0
```
Comment (BenMukebo) [inline app/controllers/posts_controller.rb]: - [ ] Great job so far 👏🏻  However, According to the project requirement 👇🏻 , A post should be created on behalf of the app's currently logged-in user (current_user). Kindly address this 👍🏻 

Screenshot from 2022-12-17 22-38-19

- The posts controller should have a create actions similar to the one shown below:

```ruby
    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @post = Post.find_by(id: params[:post_id])
+    @comment.author_id = current_user.id
+    @comment.post_id = @post.id
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: ':( Cannot Create comment retry again :('
+    end
+  end
```
Comment (BenMukebo) [inline app/controllers/comments_controller.rb:17]: - [x] Great job so far 👏🏻  However, According to the project requirement 👇🏻 , A comment should be created on behalf of the app's currently logged-in user (current_user). Kindly address this 👍🏻 


```ruby
   def create
       @comment = @current_user.comments.new(params)
    ...
  end
```

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def create
+    @post = Post.find_by(id: params[:post_id])
+    @like = @post.likes.new(author_id: current_user.id, post_id: @post)
+
+    if @like.save
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: ':( Cannot Add A Like retry again :('
+    end
+  end
```
Comment (BenMukebo) [inline app/controllers/likes_controller.rb:15]: - [x] Same thing here, A like should be created on behalf of the app's currently logged-in user (current_user). The likes controller should have a create actions similar to the one shown below:

```ruby
  def create
      @comment = @current_user.comments.new(params)
       ....
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @X-Elie-X,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (X-Elie-X) [reply app/controllers/posts_controller.rb]: Hello @BenMukebo The current user method is actually implemented. Its there inside the create method. When a transaction is made it takes the current user id. Kindly check on that from your side please.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (X-Elie-X) [reply app/controllers/comments_controller.rb:17]: Hello @BenMukebo the current user is also referenced inside the create method If you check it out on line 9 it has the current_user.id that is used to reference the current user that created the post. Kindly check that too.

**Review by DeliceLydia — APPROVED** [review]
## Hi @X-Elie-X, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**